From 95f05aa6308f7a8b22a022cd010b1aa71b6af52d Mon Sep 17 00:00:00 2001
From: Lars Kellogg-Stedman <lars@redhat.com>
Date: Thu, 21 Jun 2018 15:26:50 -0400
Subject: [PATCH 3/4] [WIP] pass list of memcache servers to
 keystone::federation::openidc

Change-Id: I58e364a3a6c0ebc7bc57ff5821ccdb882324ff81
Depends-on: I9ff976854b93cdf9ca3175d1fd39c2b268b9f7fa
(cherry picked from commit 1ecd9448a58fc898f77aede9589efc08b58fa226)
---
 manifests/profile/base/keystone.pp | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/manifests/profile/base/keystone.pp b/manifests/profile/base/keystone.pp
index eb1acf80..c51b5987 100644
--- a/manifests/profile/base/keystone.pp
+++ b/manifests/profile/base/keystone.pp
@@ -149,6 +149,11 @@
 # [*keystone_openidc_enabled*]
 #   (Optional) Enable OpenIDC federation
 #   Defaults to hiera('keystone_openidc_enabled', false)
+#
+# [*memcached_ips*]
+#   (Optional) Array of ipv4 or ipv6 addresses for memcache.
+#   Defaults to hiera('memcached_node_ips')
+#
 class tripleo::profile::base::keystone (
   $admin_endpoint_network         = hiera('keystone_admin_api_network', undef),
   $bootstrap_node                 = hiera('bootstrap_nodeid', undef),
@@ -180,6 +185,7 @@ class tripleo::profile::base::keystone (
   $keystone_enable_member         = hiera('keystone_enable_member', false),
   $keystone_federation_enabled    = hiera('keystone_federation_enabled', false),
   $keystone_openidc_enabled       = hiera('keystone_openidc_enabled', false),
+  $memcached_ips                  = hiera('memcached_node_ips')
 ) {
   if $::hostname == downcase($bootstrap_node) {
     $sync_db = true
@@ -271,7 +277,11 @@ class tripleo::profile::base::keystone (
     }
 
     if $keystone_openidc_enabled {
-      include ::keystone::federation::openidc
+      $memcached_servers = suffix(any2array(normalize_ip_for_uri($memcached_ips)), ':11211')
+
+      class { '::keystone::federation::openidc':
+        memcached_servers => $memcached_servers,
+      }
     }
   }
 
-- 
2.18.0.rc2

